lib/pages/theme_page/theme_page.dart

```diff
@@ -0,0 +1,65 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:seg_coursework_app/models/custom_theme_details.dart';
+import 'package:seg_coursework_app/widgets/theme_grid.dart';
+import '../../themes/themes.dart';
+import 'customize_theme_page.dart';
+
+class ThemePage extends StatefulWidget {
+  const ThemePage({super.key});
+
+  @override
+  State<ThemePage> createState() => _ThemePageState();
+}
+
+///The page for the user to select a theme to be displayed.
+class _ThemePageState extends State<ThemePage> {
+
+  ///This function is called whenever the save button in the customize theme page is pressed and allows
+  ///the theme grid to show the recently added theme.
+  void updateThemeList(List<CustomThemeDetails> themeList, CustomTheme themeNotifier)
+  {
+    setState(() {
+      themeList = themeNotifier.getThemes();
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final themeNotifier = Provider.of<CustomTheme>(context);
+    final List<CustomThemeDetails> themeList = themeNotifier.getThemes();
+
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text("Themes"),
+        leading: IconButton(
+          key: const Key("backButton"),
+          icon: const Icon(Icons.arrow_back),
+          onPressed: () => Navigator.pop(context),
+          tooltip: "Back",
+        ),
+        actions: [
+          IconButton(
+            key: Key("addThemeButton"),
+            tooltip: "Add a new theme",
+            icon: Icon(Icons.add),
+            onPressed: () {
+              Navigator.push(
+                context,
+                MaterialPageRoute(
+                  builder: (context) => CustomizeThemePage(themeList: themeList, updateThemeList: updateThemeList),
+                ),
+              );
+            }, 
+          ),
```

Comment:
Instead of having this add theme button here. Could be better to have a round button as the one in the visual timetable at the bottom right corner with a + icon on it. Would recommend adding more padding to the right and bottom (especially to the right) and increase the size of it as well. 

<img width="1512" alt="Screenshot 2023-03-02 at 18 24 29" src="https://user-images.githubusercontent.com/99401670/222489407-cca4eecc-6bfa-48e6-88ba-03cecd27553f.png">